diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/ClassLikeStructuresOrderRule.kt

```diff
@@ -88,28 +88,35 @@ class ClassLikeStructuresOrderRule(private val configRules: List<RulesConfig>) :
                 } ?: false
             }
             .let { it.first.toMutableList() to it.second.toMutableList() }
-        val companion = node.getChildren(TokenSet.create(OBJECT_DECLARATION))
-            .find { it.findChildByType(MODIFIER_LIST)?.findLeafWithSpecificType(COMPANION_KEYWORD) != null }
-        val blocks = Blocks(AllProperties(loggers, constProperties, properties, lateInitProperties),
-            initBlocks, constructors, methods, usedClasses, listOfNotNull(companion).toMutableList(),
+        val (companionObject, objects) = node.getChildren(TokenSet.create(OBJECT_DECLARATION))
+            .partition { it.findChildByType(MODIFIER_LIST)?.findLeafWithSpecificType(COMPANION_KEYWORD) != null }
+        val blocks = Blocks(AllProperties(loggers, constProperties, properties, lateInitProperties), objects.toMutableList(),
+            initBlocks, constructors, methods, usedClasses, companionObject.toMutableList(),
             unusedClasses)
-
-        blocks
             .allBlockFlattened()
-            .reversed()
-            .handleIncorrectOrder(blocks::getSiblingBlocks) { astNode, beforeThisNode ->
-                WRONG_ORDER_IN_CLASS_LIKE_STRUCTURES.warnAndFix(configRules, emitWarn, isFixMode, "${astNode.elementType}: ${astNode.text}", astNode.startOffset, astNode) {
-                    val replacement = node.moveChildBefore(astNode, beforeThisNode, true)
-                    replacement.oldNodes.forEachIndexed { idx, oldNode ->
+            .map { astNode ->
+                listOf(astNode) + astNode.siblings(false).takeWhile { it.elementType == WHITE_SPACE || it.isPartOfComment()}.toList()
+            }
+
+        val classChildren = node.children().filter { it.elementType in childrenTypes }.toList()
+        if (classChildren != blocks.map { it.first() }) {
+            blocks.forEachIndexed { index, listOfChildren ->
+                val astNode = listOfChildren.first()
+                if (astNode != classChildren[index]) {
```

Comment:
You could have this predicate in `filterIndexed` before `forEach` to avoid additional block depth level.